buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/shared-configuration.kt

```diff
@@ -106,11 +107,20 @@ fun Project.integrationTestUsesSampleDir(vararg sampleDirs: String) {
 }
 
 
+/**
+ * kotlin-dsl-plugins integration test depends on jars published to local repository
+ */
 fun Project.integrationTestUsesKotlinDslPlugins() {
     tasks.withType<IntegrationTest>().configureEach {
-        inputs.dir(project(":kotlinDslPlugins").buildDir.resolve("repository"))
-            .withPropertyName("localRepository")
+        inputs.files(callable {
+            project(":kotlinDslPlugins")
+                .fileTree("build/repository")
+                .matching { include("**/*.jar") }
+                .files
+                .toSortedSet()
```

Comment:
With this filtering, will the pom/xml files still be uploaded to the distributed test agents? (we don't need them to be defined as input, but we need some version of them for the tests to work)

Reply:
Ah good point.

Reply:
One option would be to post process the metadata xml files in the test fixture (?) to remove the timestamp. It should be the only problem. And it is not needed when we read the file during dependency resolution.